app/src/androidTest/java/org/mozilla/focus/activity/WebContextMenuTest.java

```diff
@@ -105,43 +132,122 @@ protected void afterActivityFinished() {
-        final DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
-        final int displayWidth = displayMetrics.widthPixels;
-        final int displayHeight = displayMetrics.heightPixels;
-        onView(withId(R.id.main_content)).check(matches(isDisplayed())).perform(AndroidTestUtils.clickXY(displayWidth / 2, displayHeight / 2, Tap.LONG));
+        loadTestWebsiteAndOpenContextMenu();
+
+        // By default Espresso Intents does not stub any Intents. Stubbing needs to be setup before
+        // every test run. In this case all external Intents will be blocked.
+        intending(not(isInternal())).respondWith(new Instrumentation.ActivityResult(Activity.RESULT_OK, null));
+
+        // Click "Share link"
+        onView(allOf(withText(R.string.contextmenu_link_share), isDisplayed())).perform(click());
+
+        // Check if intent is sent
+        intended(allOf(hasAction(Intent.ACTION_CHOOSER),
+                hasExtra(is(Intent.EXTRA_INTENT),
+                        allOf( hasAction(Intent.ACTION_SEND),
+                                hasExtra(Intent.EXTRA_TEXT, getLinkUrl())))));
+    }
+
+    @Test
+    public void copyLinkAddress() {
+
+        loadTestWebsiteAndOpenContextMenu();
+
+        // Click "Copy link address"
+        onView(allOf(withText(R.string.contextmenu_link_copy), isDisplayed())).perform(click());
+
+        // Get clip data form clip board
+        final ClipboardManager clipboard = (ClipboardManager)
+                intentsTestRule.getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+        final ClipData clipData = clipboard.getPrimaryClip();
+        Assert.assertNotNull(clipData);
+
+        final ClipData.Item clipItem = clipData.getItemAt(0);
+        Assert.assertNotNull(clipItem);
+
+        // Check if uri is matched
+        Assert.assertEquals(clipItem.getUri(), Uri.parse(getLinkUrl()));
+    }
+
+    @Test
+    public void shareImage_sendShareIntent() {
+
+        loadTestWebsiteAndOpenContextMenu();
+
+        // By default Espresso Intents does not stub any Intents. Stubbing needs to be setup before
+        // every test run. In this case all external Intents will be blocked.
+        intending(not(isInternal())).respondWith(new Instrumentation.ActivityResult(Activity.RESULT_OK, null));
+
+        // Click "Share image"
+        onView(allOf(withText(R.string.contextmenu_image_share), isDisplayed())).perform(click());
+
+        // Check if intent is sent
+        intended(allOf(hasAction(Intent.ACTION_CHOOSER),
+                hasExtra(is(Intent.EXTRA_INTENT),
+                        allOf( hasAction(Intent.ACTION_SEND),
+                                hasExtra(Intent.EXTRA_TEXT, getLinkUrl())))));
+    }
+
+    @Test
+    public void copyImageAddress() {
+
+        loadTestWebsiteAndOpenContextMenu();
+
+        // Click "Copy image address"
+        onView(allOf(withText(R.string.contextmenu_image_copy), isDisplayed())).perform(click());
```

Comment:
I have a problem running this test randomly.
The test stops at this line when the test runs for the first time.
There's a new shared preference "PREF_KEY_INT_ONBOARDING_VERSION" to display new onboarding for theme-changing feature.
You may need to add that to beforeTest

Reply:
Already disable color theme notification before running test.